packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/CozyMarkerBuilder.java

```diff
@@ -136,12 +139,41 @@ private Bitmap addBubbleMarkerText(String text) {
         return marker;
     }
 
+    private Bitmap addRoundedMarkerText(String text) {
+        Rect rect = new Rect();
+        bubbleTextPaint.getTextBounds(text, 0, text.length(), rect);
+        int ovalMarkerSize = size / 2;
+        int minWidth = rect.width() > ovalMarkerSize ? rect.width() : ovalMarkerSize;
+
+        int width = minWidth + padding + shadowSize;
+        int height = rect.height() + padding + shadowSize;
+
+        RectF shadow = new RectF(0, 0, width, height);
+        RectF shape = new RectF(shadowSize, shadowSize, width - shadowSize, height - shadowSize);
+
+        Bitmap marker = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+
+        double density = Resources.getSystem().getDisplayMetrics().density;
+        int borderRadius = (int) (15 * density);
+        Canvas canvas = new Canvas(marker);
+        canvas.drawRoundRect(shadow, borderRadius * 5, borderRadius * 5, getShadowPaint());
+        canvas.drawRoundRect(shape, borderRadius, borderRadius, getBackgroundColor());
+
+        float dx = (width / 2f) - (rect.width() / 2f) - rect.left;
+        float dy = (height / 2f) + (rect.height() / 2f) - rect.bottom;
+
+        canvas.drawText(text, dx, dy, bubbleTextPaint);
```

Comment:
Do you think we could break down this method (and possibly the other ones) into smaller ones so this method is easier to reason about? For instance:
```java
drawTopSide();
drawRightSide();
...
```

Reply:
Nice idea. Will do!

Reply:
I'll try it out and let you know how it goes!